ert_gui/tools/plot/plot_api.py

```diff
@@ -1,115 +1,189 @@
+                return e
+        return None
 
-    def _get_all_cases(self) -> List[str]:
+    def _get_all_cases(self) -> List[dict]:
         if self._all_cases is not None:
-            return list(self._all_cases.keys())
+            return self._all_cases
 
-        self._all_cases = {}
+        self._all_cases = []
         with Storage.session() as client:
             try:
                 response = client.get("/experiments")
                 experiments = response.json()
-                for experiment in experiments:
-                    for ensemble_id in experiment["ensemble_ids"]:
-                        response = client.get(f"/ensembles/{ensemble_id}")
-                        response_json = response.json()
-                        case_name = response_json["userdata"]["name"]
-                        self._all_cases[case_name] = ensemble_id
-                return list(self._all_cases.keys())
-            except Exception as e:
-                logging.exception(f"Could not retrieve case information! {str(e)}")
-                return []
+                experiment = experiments[0]
+                for ensemble_id in experiment["ensemble_ids"]:
+                    response = client.get(f"/ensembles/{ensemble_id}")
+                    response_json = response.json()
+                    case_name = response_json["userdata"]["name"]
+                    self._all_cases.append(
+                        dict(
+                            name=case_name,
+                            id=ensemble_id,
+                            hidden=case_name.startswith("."),
+                        )
+                    )
+                return self._all_cases
+            except (httpx.RequestError, IndexError) as exc:
+                logging.exception(exc)
+                raise exc
+
+    def _get_experiment(self) -> dict:
+        with Storage.session() as client:
+            try:
+                response: requests.Response = client.get("/experiments")
+                response_json = response.json()
+                return response_json[0]
+            except (httpx.RequestError, IndexError) as exc:
+                logger.exception(exc)
+                raise exc
+
+    def _get_ensembles(self, experiement_id) -> List:
+        with Storage.session() as client:
+            try:
+                response: requests.Response = client.get(
+                    f"/experiments/{experiement_id}/ensembles"
+                )
+                response_json = response.json()
+                return response_json
+            except httpx.RequestError as exc:
+                logger.exception(exc)
+                raise exc
 
     def all_data_type_keys(self):
-        """Returns a list of all the keys except observation keys. For each key a dict is returned with info about
+        """Returns a list of all the keys except observation keys.
+
+        The keys are only read from the first ensemble as they should be the same
```

Comment:
Not sure this assumption is true. I remember a discussion about a possible edge case where some ensembles might have either some extra keys or some keys missing from the other ensembles.

Reply:
@berland do you know this?

Reply:
Particularly in prediction, the set of Eclipse summary keys easily differs between realizations (and then also ensembles). In an assisted history match setting, it is thinkable, but not common.